relay-server/src/services/processor.rs

```diff
@@ -1839,20 +1844,26 @@ impl EnvelopeProcessorService {
                     namespace: Some(namespace),
                 };
 
-                (!self.rate_limit_buckets(item_scoping, &buckets, quotas, mode, rate_limiter))
-                    .then_some(buckets)
+                (!self.rate_limit_buckets(
+                    item_scoping,
+                    &buckets,
+                    dynamic_quotas.iter(),
+                    mode,
+                    rate_limiter,
+                ))
+                .then_some(buckets)
             })
             .flatten()
             .collect()
     }
 
     /// Returns `true` if the batches should be rate limited.
     #[cfg(feature = "processing")]
-    fn rate_limit_buckets(
-        &self,
+    fn rate_limit_buckets<'a>(
```

Comment:
Don't think this is necessary

Reply:
the lifetime?

otherwise i get this: `anonymous lifetimes in `impl Trait` are unstable`

Reply:
Now you do, it used to be a concrete type, which it still should be here to prevent mistakes, at this point we always want combined quotas, the type system should enforce that.